src/pipeline.ts

```diff
@@ -639,7 +643,16 @@ export class GitHubWorkflow extends PipelineBase {
     };
   }
 
-  private stepsToConfigureAws(openId: boolean, { region, assumeRoleArn }: { region: string; assumeRoleArn?: string }): github.JobStep[] {
+  private ghaStageRoleArn(): string {
+    const stageRoleArn = Object.entries(this.stackProperties)[0][1].role;
```

Comment:
this is excessively brittle... `stackProperties` should be accessed by `stackArtifactId`.

Reply:
Sorry. I really struggled with this. I was doing as you suggested in `jobForDeploy()`:

    const gitHubActionsStageRoleArn = this.stackProperties[stack.stackArtifactId]?.role

But in `jobForAssetPublish()`, I didn't have access to `stack`, because it didn't seem available to pass in here: https://github.com/cdklabs/cdk-pipelines-github/blob/main/src/pipeline.ts#L345

Any ideas for how I can solve this?

Reply:
I've looked into this quite a bit more, and I can't figure out a nice way to work out which `Stage` corresponds to which "asset publish" step. Do you have any ideas here?

Reply:
I've tried lots of different things here, but still can't figure out how to link the `publish` step to the correct stage in `jobForAssetPublish`. So I can't figure out which role should be used for the publish step.

Reply:
Hi @danieljamesscott. This could be a problem. You're right, `jobForAssetPublish` doesn't have a concept of `Stage`, because it is stage-agnostic. What's happening is that the assets for all stages are combined into the asset publishing step, which is really supposed to be abstracted from the user. That's the benefit of having a single "pipeline" role that has the correct permissions just like in aws-cdk-lib/pipelines.

Having thought this through a bit, I'm a little less groovy of the idea of natively supporting a role per stage. This looks like it will introduce more problems than it solves. If we did support this, we'd need to introduce a new property like `assetPublishingRoleArn` and then we're not really solving the problem you put forth -- somewhere one of your AWS accounts is going to have to own the role for asset publishing.

Ideally, we introduce some sort of escape hatching mechanism that allows users to go off the beaten path and do what they want. Let's take this conversation back to the issue and see if we can come up with a design that makes sense.